File: qc-application/src/main/java/com/eager/questioncloud/application/api/user/profile/UserProfileService.java
package com.eager.questioncloud.application.api.user.profile;

import com.eager.questioncloud.domain.user.User;
import com.eager.questioncloud.domain.user.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class UserProfileService {
    private final UserRepository userRepository;

    public void updateUserInformation(User user, String name, String profileImage) {
        user.updateUserInformation(name, profileImage);
        userRepository.save(user);
    }
}

File: qc-core/qc-user/qc-user-api/src/main/kotlin/com/eager/questioncloud/user/profile/service/UserProfileService.kt
package com.eager.questioncloud.user.profile.service

import com.eager.questioncloud.user.common.implement.UserFinder
import com.eager.questioncloud.user.dto.MyInformation
import com.eager.questioncloud.user.profile.implement.UserProfileUpdater
import org.springframework.stereotype.Service

@Service
class UserProfileService(
    private val userFinder: UserFinder,
    private val userProfileUpdater: UserProfileUpdater
) {
    fun updateUserInformation(userId: Long, name: String, profileImage: String) {
        userProfileUpdater.updateUserInformation(userId, name, profileImage)
    }
    
    fun getMyInformation(userId: Long): MyInformation {
        val user = userFinder.getById(userId)
        return MyInformation.from(user)
    }
}
